### PR TITLE
Rename debug_log to tracelog

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
debug_log was a bad name... tracelog is slightly better...
This PR does not add any new functionality, just a strict name change.
This is preparation for some PRs which finish this test feature, namely:
- extend to cover all of the wandb service mode
- add a new output format that is more extensible
- add tools to convert tracelog output into other traceing / telemetry formats

Testing
-------
Added a slightly better test (using logger mode)... once the tools are checked in we can add a better test (validate state transitions, etc)

Checklist
-------
- Name PR "[WB-NNNN][WB-MMMM] Add support for..." similar to entries in CHANGELOG.md
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/sdk/wandb_settings.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ class Settings:
     _cli_only_mode: bool  # Avoid running any code specific for runs
     _config_dict: Config
     _cuda: str
-    _debug_log: str
+    _tracelog: str
```
Comment (dmitryduev) [inline wandb/sdk/wandb_settings.py:350]: Please keep it in the alphabetic order :)

File: functional_tests/debug/02-tracelog-logger.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+trace_log_str = "TRACELOG(1)"
+debug_log_internal_count = 0
+with open(debug_log_user) as f:
+    debug_log_user_count = f.read().count(trace_log_str)
+with open(debug_log_internal) as f:
+    debug_log_internal_count = f.read().count(trace_log_str)
+print(f"Log user count: {debug_log_user_count}")
+print(f"Log internal count: {debug_log_internal_count}")
+assert debug_log_user_count > 10
+assert debug_log_internal_count > 10
```
Comment (dmitryduev) [inline functional_tests/debug/02-tracelog-logger.py:47]: Maybe rename all of these to tracelog* as well?

**Review by dmitryduev — APPROVED** [review]

File: functional_tests/debug/02-tracelog-logger.py (hunk 2 above)
Comment (raubitsj) [reply functional_tests/debug/02-tracelog-logger.py:47]: no..  those are actually debug_log... like debug.log and debug-internal.log
which is one of the reasons i wanted to rename this.

File: wandb/sdk/wandb_settings.py (hunk 1 above)
Comment (raubitsj) [reply wandb/sdk/wandb_settings.py]: oops... of course.. fixed.

**Pull request merged**